### PR TITLE
VideoSoftware: make use of Clamp()

### PR DESCRIPTION
**Pull request by Tilka**

### CODE REVIEW

File: Source/Core/VideoBackends/Software/Tev.cpp [hunk 1]
```diff
@@ -776,9 +777,7 @@ void Tev::Draw()
       // Based on that, choose the index such that points which are far away from the z-axis use the
       // 10th "k" value and such that central points use the first value.
       float floatindex = 9.f - std::abs(offset) * 9.f;
-      floatindex = (floatindex < 0.f) ?
-                       0.f :
-                       (floatindex > 9.f) ? 9.f : floatindex;  // TODO: This shouldn't be necessary!
+      floatindex = MathUtil::Clamp(floatindex, 0.f, 9.f);  // TODO: This shouldn't be necessary!
```
Comment (BhaaLseN) [inline Source/Core/VideoBackends/Software/Tev.cpp]: Do you know why "This shouldn't be necessary"?
And if so, can you update (or even fix) that TODO to give the reason?
I can see this being assumed in the 0..9 range, but is that clamp because the float might be inaccurately representing the values?

Comment (Tilka) [reply Source/Core/VideoBackends/Software/Tev.cpp]: No idea, the comment was originally introduced in 0994a5828d5a4528aa65f7dd7c6733a33445a675.

Comment (phire) [reply Source/Core/VideoBackends/Software/Tev.cpp]: The line above makes an assumption that bpmem.fogRange.Base.Center will actually be in the center of the screen.

There is nothing (in dolphin) stopping the game setting the center to 0 (which would result in a negative offset for pixels in the left 342 pixels) or setting the center to the maximum of 1023 (which would result in most pixels on the right hand side, depending on xfmem.viewport.wd, having values larger than 1.0)

What happens on actual hardware, who knows. It doesn't look like any hardware tests have been done for this edge case.
